### PR TITLE
net/nanocoap: allow user to configure macros

### PR DESCRIPTION
**Pull request by kb2ma**
### Contribution description
Allows the user to externally specify a value for several macros in nanocoap used to establish maximum sizes. Also adds a documentation group for these user configurable macros, per #10566. 

### Testing procedure
Compile the documentation. You should see:

- a new configuration group titled, "Nanocoap compile configurations"
- a new module within "Nanocoap", also titled "Nanocoap compile configurations"

### Issues/PRs references
Partially implements #10566. Depends on #10687.

### CODE REVIEW

File: sys/include/net/nanocoap.h [hunk 1]
```diff
@@ -178,20 +178,38 @@ extern "C" {
...
+ */
+#ifndef NANOCOAP_URI_MAX
 #define NANOCOAP_URI_MAX            (64)
-#define NANOCOAP_BLOCK_SIZE_EXP_MAX  (6)  /**< Maximum size for a blockwise
-                                            *  transfer as power of 2 */
-/** @} */
+#endif
 
-#ifdef MODULE_GCOAP
-#define NANOCOAP_URL_MAX        NANOCOAP_URI_MAX
```
[the remark this answers is not in the document]
Comment (kb2ma) [reply sys/include/net/nanocoap.h]: Moved removal to #10687, and rebased this PR on it.

File: sys/include/net/nanocoap.h [hunk 2]
```diff
@@ -178,20 +178,38 @@ extern "C" {
 #define COAP_FORMAT_NONE        (UINT16_MAX)
 
 /**
- * @name    Nanocoap specific maximum values
+ * @defgroup net_nanocoap_conf    Nanocoap compile configurations
+ * @ingroup  config
```
[the remark this answers is not in the document]
Comment (kb2ma) [reply sys/include/net/nanocoap.h]: Fixed in da68c95.

**Comment by kb2ma** [comment]
All comments addressed.

**Comment by kb2ma** [comment]
Squashed. Probably makes sense to merge #10687 first. Sorry if that wasn't clear. I just added the appropriate label a few minutes ago.

**Comment by miri64** [comment]
#10687 was merged this. Let's murdock check this PR.

**Pull request merged**